rapidxml.hpp

```diff
@@ -1723,6 +1741,11 @@ namespace rapidxml
                 *dest++ = *src++;
 
             }
+            if(shorter) {
+                // Add terminating zero after decoded shorter value 2
+                if (!(Flags & parse_no_string_terminators))
+                    *dest = 0;
```

Comment:
Can you describe the issue this is supposed to fix?

Reply:
I added some tests for character references in 3ade9030f73b03bd648383da0ea0402b92da4e97.
To me, everything (nul-terminated strings & length) seems to be just fine. Do you have a test-case that shows your issue?

Reply:
Found another bad feature - default unescape affect also ex. CR & LF (&amp;#13; &amp;#10; &amp;#xA; &amp;#xD;) so in multi-platform environment you cannot process XML without unrecoverable CR/LF mismatch sometimes...